### DISCUSSION TITLE
GoRouter like Shell Routing feature

### DISCUSSION THREAD

**Discussion by rakhaxor** [nylo-core/nylo#101] [Ideas] [answered]
I was building a persistent bottom navigation bar where all the pages keep their state while navigating. I built it using AutomaticKeepAliveClientMixin and it's working fine. Now I wanted to create subpages under the First page (Home screen). And I want the app to keep showing the bottom navigation bar when I go from Home to it's subpage i.e. some detail screen. I am looking for a solution. I found that GoRouter provides it out of the box and I wouldn't need to use AutomaticKeepAliveClientMixin but it would need me to change the whole routing which I don't want. I also don't want to create another MaterialApp within a MaterialApp to do it. I thought of sharing this idea.

If I can get a solution to the said problem,  I would really appreciate it.

Thank you!

**Comment by agordn52** [comment]
Hi @rakhaxor,

Thanks for sharing this.

I'll investigate this to see if there's an easy way to implement this within the framework.
If you have any ideas for a PR feel free to send our idea through too.

Thanks

**Comment by rakhaxor** [comment]
Hello my friend! Hope you are well. I was wondering if there's any solution yet. Excuse me for late response, I am not experienced in Flutter so can't really code it myself for the PR.

**Comment by agordn52** [comment]
Hi @rakhaxor,

Sorry for the late response and thanks for the update.
At the moment, I can't think of an easy way to implement this into the framework.

If I have any ideas, I'll let you know.

**Comment by rakhaxor** [comment, marked as answer]
I actually achieved what I wanted to do with a library. Here's the link in case you would like to see what they are doing. It has nothing to do with routing but achieves what's rather a common requirement for apps.

https://pub.dev/packages/persistent_bottom_nav_bar

**Reply by agordn52** [reply]
Hi @rakhaxor,

Thanks for sharing this. 

I'll mark your response as the answer for Devs who need this in their projects too 👍
